test/ctia/test_helpers/core.clj

```diff
@@ -281,11 +281,21 @@
                   time/format-date-time)))
 
 (defn encode [s]
-  {:pre [(string? s) (seq s)]}
+  (assert (string? s)
```

Comment:
I've always found preconditions to:
1) be a complicated way to basically just do an assert
2) generate not-so-hot error messages.